docs/configuration.rst

```diff
@@ -149,6 +149,11 @@ This defines the version that will be pulled from just the Open edX platform git
 
 By default, there are 2 `uwsgi worker processes <https://uwsgi-docs.readthedocs.io/en/latest/Options.html#processes>`__ to serve requests for the LMS and the CMS. However, each worker requires upwards of 500 Mb of RAM. You should reduce this value to 1 if your computer/server does not have enough memory.
 
+- ``OPENEDX_LMS_CELERY_WORKERS`` (default: ``"0"``)
+- ``OPENEDX_CMS_CELERY_WORKERS`` (default: ``"0"``)
```

Comment:
Adding new configuration settings to Tutor core is a personal trigger of mine :firecracker: Do we really want to make changes to the default production values? If yes, can we:

1. propose better defaults?
2. make these custom changes possible via a patch instead of two new configuration settings?

Reply:
Yeah, I understand the reluctance to add new config items.  It's just that in this case, it was the most straightforward way to achieve what I was after.  There's precedent, too: `OPENEDX_LMS_UWSGI_WORKERS` is there for very similar reasons.

Regarding the defaults, I'm not actually changing them: I'm just making them explicit, where before they were explicit.  (The implicit default is to scale the workers to however many CPUs you have, and that's what "0" means.)

As for using patches, I wouldn't mind except for the fact that, as mentioned above, this is just doing what `OPENEDX_LMS_UWSGI_WORKERS` does, except for Celery workers.  If we have that configuration, I don't see why we shouldn't have this one.

All of this said, I really like the idea of changing certain things automatically for development environments, whether they have corresponding config items or not.  For instance, after I issued this PR it came to my attention that Tutor's [importing `*` from devstack.py](https://github.com/overhangio/tutor/blob/master/tutor/templates/apps/openedx/settings/lms/development.py#L3) for the development settings, and that means that we aren't using workers at all!  (See https://github.com/openedx/edx-platform/blob/master/lms/envs/devstack.py#L35.)  So why is `tutor dev` even firing up workers?

In any case, the latter sounds like it warrants a separate PR.  My question regarding _this_ PR, though, is whether we do or do not want `OPENEDX_LMS_CELERY_WORKERS`.  It might not make sense to change this in a Kubernetes setting, but I'm willing to defend that it _does_ on any Docker deployment where you have more CPUs than you have RAM (so to speak).

Reply:
How about this for not starting workers at all in dev mode?  https://github.com/overhangio/tutor/pull/1041

The question remains whether we still want to let people configure the number of Celery workers manually.  (I say we let them.)

Reply:
I love the fact that we can disable workers in dev. I commented on #1041. 

Let's now focus on the possibility to customize the number of celery runners. I agree that this would be a useful feature. If we really have to, we'll introduce new configuration values, but I'd like to see if we can avoid it. For instance, could we avoid that by creating a celery [config file](https://docs.celeryq.dev/en/stable/userguide/configuration.html#id11)? This file would include a `{{ patch("edx-platform-celery-config") }}` statement. That way, we wouldn't have to create new configuration settings for every celery parameter.

Reply:
@regisb
Agree! There should be an option to disable workers in dev, it could be disabled by default on dev mode. Personally, I like of having a config file with a patch. By default the config file should have the minimum config to start.
Nevertheless, I feel that won't resolve every additional configuration:
- https://github.com/celery/celery/issues/2566
- https://docs.celeryq.dev/en/stable/reference/cli.html
- https://docs.celeryq.dev/en/stable/userguide/configuration.html this won't have a way to disable gossip, mingle, heartbeat and configure (vertical) autoscale.

Reply:
Please see my answer here: https://github.com/overhangio/tutor/issues/1126#issuecomment-2385620266
I propose that all remaining comments are made on issue #1126